fgrpc/grpcrunner_test.go

```diff
@@ -30,34 +30,168 @@ import (
 	"google.golang.org/grpc/health/grpc_health_v1"
 )
 
+var (
+	// Generated from "make cert"
+	caCrt  = "../cert-tmp/ca.crt"
+	svrCrt = "../cert-tmp/server.crt"
+	svrKey = "../cert-tmp/server.key"
+	// used for failure test cases
+	failCrt = "../missing/cert.crt"
+	failKey = "../missing/cert.key"
+)
+
 func TestGRPCRunner(t *testing.T) {
 	log.SetLogLevel(log.Info)
-	port := PingServer("0", "bar", 0)
-	destination := fmt.Sprintf("localhost:%d", port)
+	iPort := PingServer("0", "", "", "bar", 0)
+	iDest := fmt.Sprintf("localhost:%d", iPort)
+	sPort := PingServer("0", svrCrt, svrKey, "bar", 0)
+	sDest := fmt.Sprintf("localhost:%d", sPort)
 
-	opts := GRPCRunnerOptions{
-		RunnerOptions: periodic.RunnerOptions{
-			QPS:        100,
-			Resolution: 0.00001,
+	tests := []struct {
+		name       string
+		runnerOpts GRPCRunnerOptions
+		expect     bool
+	}{
+		{
+			name: "insecure runner with payload",
+			runnerOpts: GRPCRunnerOptions{
+				RunnerOptions: periodic.RunnerOptions{
```

Comment:
NeedFix: there is a lot of copy paste of this section, why not have the runner options in a variable ?

Reply:
I guess I find table tests easier to read when testing multiple test cases. WOuld you rather I follow the existing pattern for the test? Something like this:

```
func TestGRPCRunner(t *testing.T) {
	log.SetLogLevel(log.Info)
	iPort := PingServer("0", "", "", "bar", 0)
	iDest := fmt.Sprintf("localhost:%d", iPort)
	sPort := PingServer("0", svrCrt, svrKey, "bar", 0)
	sDest := fmt.Sprintf("localhost:%d", sPort)
	ro := periodic.RunnerOptions{
		QPS:        100,
		Resolution: 0.00001,
	}

	opts := GRPCRunnerOptions{
		RunnerOptions: ro,
		Destination: iDest,
		Profiler:    "test.profile",
	}
	res, err := RunGRPCTest(&opts)
	if err != nil {
		t.Error(err)
		return
	}
	totalReq := res.DurationHistogram.Count
	ok := res.RetCodes[grpc_health_v1.HealthCheckResponse_SERVING]
	if totalReq != ok {
		t.Errorf("Mismatch between requests %d and ok %v", totalReq, res.RetCodes)
	}
	// Test a grpc runner with TLS enabled.
	opts.Destination = sDest
	opts.CACert = caCrt
	res, err = RunGRPCTest(&opts)
	if err != nil {
		t.Error(err)
		return
	}
	totalReq = res.DurationHistogram.Count
	ok = res.RetCodes[grpc_health_v1.HealthCheckResponse_SERVING]
	if totalReq != ok {
		t.Errorf("Mismatch between requests %d and ok %v", totalReq, res.RetCodes)
	}
}
```

Keep in mind the above example is only exercising 2 test cases.